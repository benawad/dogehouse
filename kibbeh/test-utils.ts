import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
