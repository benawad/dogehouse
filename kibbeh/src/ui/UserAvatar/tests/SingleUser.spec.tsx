import React from "react";

import { render } from "../../../../test-utils";
import { SingleUser } from "../";

describe("UserAvatar", () => {
  describe("SingleUser", () => {
    const src = "";
    it("should render correctly", () => {
      const { getByTestId } = render(<SingleUser src={src} size="default" />);
      const avatar = getByTestId("single-user-avatar");

      expect(avatar).toBeVisible();
    });

    it("should show the online indicator is the isOnline prop is ture", () => {
      const { getByTestId } = render(
        <SingleUser src={src} size="default" isOnline />
      );
      const onlineIndicator = getByTestId("online-indictor");

      expect(onlineIndicator).toBeVisible();
    });

    it("should match snapshot", () => {
      const { getByTestId } = render(<SingleUser src={src} size="default" />);
      const avatar = getByTestId("single-user-avatar");

      expect(avatar).toMatchSnapshot();
    });
  });
});
