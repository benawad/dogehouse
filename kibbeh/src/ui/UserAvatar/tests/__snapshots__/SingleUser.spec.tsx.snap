// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar SingleUser should match snapshot 1`] = `
<div
  class="relative inline-block "
  data-testid="single-user-avatar"
  style="width: 80px; height: 80px;"
>
  <img
    alt="your-avatar"
    class="rounded-full w-full h-full object-cover "
    src=""
  />
</div>
`;
